mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -48,8 +48,11 @@ private[util] sealed trait BaseReadWrite {
   /**
    * Returns the user-specified SQL context or the default.
    */
-  protected final def sqlContext: SQLContext = optionSQLContext.getOrElse {
-    SQLContext.getOrCreate(SparkContext.getOrCreate())
```

Comment:
This is just a workaround for the issue. User can use `.context(hiveContext)` to specify a context to use.